app/assets/javascripts/discourse/app/templates/modal/create-account.hbs

```diff
@@ -49,10 +63,17 @@
                   <td><label>{{i18n "user.username.instructions"}}</label></td>
                 </tr>
 
-                {{#if fullnameRequired}}
+                {{#if siteSettings.enable_names}}
                   <tr class="input">
                     <td class="label">
-                      <label for="new-account-name">{{i18n "user.name.title"}}</label>
+                      <label for="new-account-name">
+                        {{i18n "user.name.title"}}
+                        {{#if siteSettings.full_name_required}}
+                          {{~#if userFields~}}
```

Comment:
Does removing whitespace only in the inner `if` achieve what you need or should the outer one have `~` too?